blocks/blocks.js

```diff
@@ -359,18 +368,242 @@ class Blocks extends Newsletter {
+      script.resize()
+      return script
+    } else if (data.indices) {
+      // Probably means the block was dragged out from a script such that the
+      // old script still remains.
+      const [workspace, scriptIndex, ...indices] = data.indices
+      let component = workspace.scripts[scriptIndex]
+      for (const index of indices) {
+        if (component instanceof Input) {
+          component = component.getValue()
+          if (!(component instanceof Block)) {
+            throw new Error('hwat. The indices point to an input that does not hold a block.')
+          }
+        }
+        if (typeof index === 'number') {
+          component = component.components[index]
+        } else {
+          component = component.getParamComponent(index)
+        }
+      }
+      const script = this.createScript()
+      if (component instanceof Input) {
+        const block = component.getValue()
+        if (!(block instanceof Block)) {
+          throw new Error('hwat. Indices point to an input that does not hold a block.')
+        }
+        component.insertBlock(null)
+        component.resize()
+        script.add(block)
+        return script
+      } else {
+        // `component` now is the block that is clicked on to drag out, so
+        // it and its younger siblings will be deported. Not all though if this
+        // is a reverse of block stack insertion.
+        const parent = component.parent
+        const { dx = 0, dy = 0 } = data
+        const { x, y } = parent.position
+        // Subtraction because this is actually the inverse of the normal operation
+        // which is done in the shove step.
+        parent.setPosition(x - dx, y - dy)
+        let index = indices[indices.length - 1]
+        if (data.branchAround) {
+          const branch = component.getParamComponent(data.branchAround)
+          while (branch.components[0]) {
+            const component = branch.components[0]
+            branch.remove(component)
+            parent.add(component, index)
+            index++
+          }
+          branch.resize()
+        }
+        let blocks = 0
+        while (blocks < blockCount) {
+          const component = parent.components[index]
+          if (!component) break
+          parent.remove(component)
+          script.add(component)
+          blocks++
+        }
+        parent.resize()
+        return script
+      }
+    } else if (data.workspace) {
+      // An entire script in a workspace
+      const script = data.workspace.scripts[data.index]
+      script.removeFromWorkspace()
+      // Should we store x/y position here? Nah
+      return script
+    } else {
+      throw new Error('wucky: Given `data` no make sense!')
+    }
+  }
+
+  /**
+   * Intended to be the reverse of whatever is done in `grabTarget`
+   */
+  shoveTarget (data, script) {
+    if (Array.isArray(data)) {
+      if (script.elem) {
+        script.destroy()
```

Comment:
Might want to check for other calls to `destroy` to make sure a destroyed component isn't redestroyed? (especially `Script`s, which autodestroy)

Reply:
Actually, this might not be necessary with [this change](https://github.com/SheepTester/ascended-cat/pull/1/files#diff-64e13fab505c4e97a902499cbaea2f44R74)? Might want to make sure it works though. Also, when does it happen that the script at this point is already destroyed?

Reply:
I removed the check lol